api/v1/clusterextensionrevision_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterExtensionRevisionSpec defines the desired state of ClusterExtensionRevision
+type ClusterExtensionRevisionSpec struct {
+	// clusterExtensionRef is a required reference to the ClusterExtension
+	// that this revision represents an available upgrade for.
+	//
+	// +kubebuilder:validation:Required
+	ClusterExtensionRef ClusterExtensionReference `json:"clusterExtensionRef"`
+
+	// version is a required field that specifies the exact version of the bundle
+	// that represents this available upgrade.
+	//
+	// version follows the semantic versioning standard as defined in https://semver.org/.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self.matches(\"^([0-9]+)(\\\\.[0-9]+)?(\\\\.[0-9]+)?(-([-0-9A-Za-z]+(\\\\.[-0-9A-Za-z]+)*))?(\\\\+([-0-9A-Za-z]+(-\\\\.[-0-9A-Za-z]+)*))?\")",message="version must be well-formed semver"
+	Version string `json:"version"`
+	// bundleMetadata contains the complete metadata for the bundle that represents
+	// this available upgrade.
+	//
+	// +kubebuilder:validation:Required
+	BundleMetadata BundleMetadata `json:"bundleMetadata"`
+
+	// availableSince indicates when this upgrade revision was first detected
+	// as being available. This helps track how long an upgrade has been pending.
+	//
+	// +kubebuilder:validation:Required
+	AvailableSince metav1.Time `json:"availableSince"`
```

Comment:
Kubernetes API conventions generally encourage time based naming to be of the format `{thing}Time`. In this case it would be something like `AvailableTime` as opposed to `AvailableSince`.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions